hydradx/model/amm/stableswap_amm.py

```diff
@@ -178,20 +178,24 @@ def spot_price(self, i: int = 1):
         return self.price_at_balance(balances, self.d, i)
 
     def sell_spot(self, tkn_sell, tkn_buy: str, fee: float = None):
-        if fee is None:
-            fee = self.calculate_fee()
         if tkn_buy not in self.liquidity or tkn_sell not in self.liquidity:
             return 0
-        else:
-            return self.price(tkn_sell, tkn_buy) * (1 - fee)
+        new_state = self.copy()
```

Comment:
I think it would be worth breaking out the specific values needed to do this calculation, rather than copying the state, for performance reasons. Buy_ and sell_spot are used potentially hundreds of times in a binary search arbitrage calculation, for example. It could be like calculate_peg_deltas, calculate_fee_from_peg_deltas, temporarily update peg deltas, calculate price, put the pegs back how they were.

Reply:
Agreed